store/views.py

```diff
@@ -532,9 +534,20 @@ def get_queryset(self):
         This viewset should return owned coupons except if
         the currently authenticated user is an admin (is_staff).
         """
-        if self.request.user.is_staff:
-            return Coupon.objects.all()
-        return Coupon.objects.filter(owner=self.request.user)
+        queryset = Coupon.objects.all()
+        if not self.request.user.is_staff:
+            # non staff can't filter on usage
+            return queryset.filter(owner=self.request.user)
```

Comment:
We should not return here I think, right now `staff` people can;t decide to filter sold coupon or not